distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso

```diff
@@ -148,9 +158,17 @@ make_join_kind_selector display=Display.Always =
 
 ## PRIVATE
    Make a join condition selector.
-make_join_condition_selector : Table -> Display -> Widget
-make_join_condition_selector table display=Display.Always =
-    col_names_selector = make_column_name_selector table display=Display.Always
+make_join_condition_selector : Table -> Display -> Function -> Widget
+make_join_condition_selector table display=Display.Always cache=Nothing =
+    right = if cache.is_nothing.not then
+        dbg
+        right_arg = cache "right"
+        if right_arg.is_nothing.not then right_arg else
+            zero_arg = cache 0
+            zero_arg
```

Comment:
What is this `zero_arg` logic? 

Does this depend if `right` was passed as positional argument VS named?

I think we need to abstract this from users (widget developers), so that we can just fetch the cached value _by argument name_. If reordering needs to be handled, maybe we need a way to extract the ordering of arguments of a method through `Meta`?

Most importantly - we are trying to re-use our widgets for different components. The position of a given argument may change between components, so that would be problematic.

Reply:
> trying to re-use our widgets for different components. The position of a given argument may change between components

So can change the name. Reusing requires abstractions and a bit of design.

Reply:
> we need to abstract this from widget developers, so that we can just fetch the cached value by **argument name**.

Yes, you are right @radeusgd. This `zero_arg` logic is too complex. Being able to rely on argument name will be much simpler.

I just need to find out the most reliable way to associate argument name with a positional argument without actually invoking the function. Probably somewhere in [CallArgumentInfo.ArgumentsMapping](https://github.com/enso-org/enso/blob/develop/engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/argument/CallArgumentInfo.java#L233) - should the argument discovery be done on the _engine side_, CCing @kustosz.

Or I can try to extract the info on the _IDE side_ via `methodCallInfo.value.methodCall`... CCing @4e6 and @farmaazon.

Reply:
Parameter info extracted on the IDE side in 654c11cc98c5874a18d491a944dcfb34d4a4431c